### PR TITLE
[WIP] Working in Progress UVP formulation

### PR DESCRIPTION
**Pull request by bodhinandach**
This is not yet to merge, just that it's easier for discussion for 2 phase 1 point MPM.

### CODE REVIEW

File: tests/particle_test.cc [hunk 1]
```diff
@@ -17,11 +17,20 @@ TEST_CASE("Particle is checked for 1D case", "[particle][1D]") {
   const unsigned Dim = 1;
```
Comment (kks32) [inline tests/particle_test.cc:17]: I'd recommend moving this to a separate two phase particle test file

File: include/particle.h [hunk 2]
```diff
@@ -179,9 +198,24 @@ class Particle : public ParticleBase<Tdim> {
     this->stress_.col(phase) = stress;
   }
 
+  //! Initial pore pressure
+  //! \param[in] pore_fluid Index corresponding to fluid phase
+  //! \param[in] pore pressure Initial pore pressure
+  void initial_pore_pressure(const unsigned pore_fluid,
```
Comment (kks32) [inline include/particle.h:204]: No need to `const` variables that are size `double` or below because it makes it less efficient. Pass those by value.

**Comment by bodhinandach** [comment]
Things to modify (feel free to add/modify):

- [ ]  Derive Particle 2 phase from the Particle class and all the necessary functions
- [ ]  Create a register
- [ ]  Derive/modify hdf5 struct including more variables specific to 2 phase particle
- [ ]  Fix 2 phase explicit strategy accordingly (see the new USL/USF flag). 
- [ ]  Fix and add testing

File: include/mpm_explicit_twophase.h [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+#include "mpm_base.h"
+
+namespace mpm {
+
+//! MPMExplicit class
+//! \brief A class that implements the fully explicit one phase mpm
+//! \details A single-phase explicit MPM
+//! \tparam Tdim Dimension
+template <unsigned Tdim>
+class MPMExplicitTwoPhase : public MPMBase<Tdim> {
```
Comment (bodhinandach) [inline include/mpm_explicit_twophase.h:13]: Hmmm, I guess 2 phase explicit should be derived from mpm_explicit, not from mpm_base to reduce code duplication. What do you think @srhgk2 @kks32?

File: include/node.tcc [hunk 4]
```diff
@@ -128,6 +130,49 @@ bool mpm::Node<Tdim, Tdof, Tnphases>::update_internal_force(
   return status;
 }
 
+//! Update mixture internal force (body force / traction force)
```
Comment (bodhinandach) [inline include/node.tcc:133]: Isn't it better to include all the specific 2 phase nodal functions to a separate .tcc file?

**Review by bodhinandach — COMMENTED** [review]
just some thoughts

**Comment by bodhinandach** [comment]
Moving this to private repo.

**Pull request closed without merging**